### ISSUE TITLE
Vue 2.5 with vuex 3.0 - TS error: All declarations of 'ComponentOptions' must have identical type parameters

### ISSUE THREAD

**Issue by DorianGrey** [vuejs/vuex#1009]
### Version
3.0.0

### Reproduction link
[https://github.com/DorianGrey/vue-ts-playground/tree/vue-2-5-update](https://github.com/DorianGrey/vue-ts-playground/tree/vue-2-5-update)
(It's not a fiddle or something similar simple, but it should be sufficient - setting up one of these for TS usage is always a bit ... unlikely.)

### Steps to reproduce
- Checkout the branch referenced above.
- Run `yarn` and `yarn build`
- The output will list a couple of errors, approx. 13. Some of them refer to augmentation errors from `vuex` and `vue-router`. Some of these errors refer to my own augmentations, just ignore them for now.
Exemplary output part:
```
 ERROR  /home/linne/Projects/vue-ts-playground/node_modules/vue-router/types/vue.d.ts
(16,13): All declarations of 'ComponentOptions' must have identical type parameters.

 ERROR  /home/linne/Projects/vue-ts-playground/node_modules/vue/types/options.d.ts
(54,18): All declarations of 'ComponentOptions' must have identical type parameters.

 ERROR  /home/linne/Projects/vue-ts-playground/node_modules/vuex/types/vue.d.ts
(9,13): All declarations of 'ComponentOptions' must have identical type parameters.
```

### What is expected?
The type augmentation that adds the `store` to the component options should not cause any build errors.

### What is actually happening?
The type augmentation actually causes build errors due to mismatching type parameters.

---
The problem is caused by these lines:
https://github.com/vuejs/vuex/blob/1234464dcded0305dba863a6486785c73f0090f7/types/vue.d.ts#L9-L11

They conflict with the declaration here:
https://github.com/vuejs/vue/blob/a433b16588412980b4ab51924de1ef91b5226f56/types/options.d.ts#L54-L59

**Comment by ktsn** [comment]
Because `vee-validate` typings and your `buefy` typing assumes old (Vue <= 2.4) typings. Please report it `vee-validate` and fix your typing to adapt Vue >= v2.5.

FYI: you should not use commonjs style import for the new typings. Just use default style import (`import Vue from 'vue'`)

**Comment by DorianGrey** [comment]
I see - I already fixed the buefy typings, just didn't commit them before. The error was a bit misleading, since it complained about literally all occurrences of `ComponentOptions` in the augmentations.
The `vee-validate` typings already got adopted, but the fix is not yet released. Seems I have to wait for that release ... 

However, thanks!
